src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/NugetUtils.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+using Microsoft.Sbom.Api.Output.Telemetry;
+using Microsoft.Sbom.Common;
+using NuGet.Configuration;
+using Serilog;
+
+namespace Microsoft.Sbom.Api.PackageDetails;
+
+/// <summary>
+/// Utilities for retrieving information from maven packages that may not be present on the buildDropPath
+/// </summary>
+public class NugetUtils : INugetUtils
+{
+    private readonly IFileSystemUtils fileSystemUtils;
+    private readonly ILogger log;
+    private readonly IRecorder recorder;
+
+    private static readonly string NugetPackagesPath = SettingsUtility.GetGlobalPackagesFolder(new NullSettings());
+
+    public NugetUtils(IFileSystemUtils fileSystemUtils, ILogger log, IRecorder recorder)
+    {
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.log = log ?? throw new ArgumentNullException(nameof(log));
+        this.recorder = recorder ?? throw new ArgumentNullException(nameof(recorder));
+    }
+
+    // Takes in a scanned component and attempts to find the associated nuspec file. If it is not found then it returns null.
+    public string GetNuspecLocation(ScannedComponent scannedComponent)
+    {
+        var nuspecLocation = string.Empty;
+
+        var componentName = scannedComponent.Component.PackageUrl?.Name.ToLower();
+        var componentVersion = scannedComponent.Component.PackageUrl?.Version;
+        var componentType = scannedComponent.Component.PackageUrl?.Type.ToLower();
+
+        if (componentType == "nuget")
+        {
+            nuspecLocation = $"{NugetPackagesPath}/{componentName}/{componentVersion}/{componentName}.nuspec";
```

Comment:
Looks like you can apply most of my comments from MavenUtils to this one as well.